pkg/apis/resource/types.go

```diff
@@ -823,104 +779,6 @@ type ResourceClaimList struct {
 
 // +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
 
-// PodSchedulingContext objects hold information that is needed to schedule
-// a Pod with ResourceClaims that use "WaitForFirstConsumer" allocation
-// mode.
-//
-// This is an alpha type and requires enabling the DRAControlPlaneController
-// feature gate.
-type PodSchedulingContext struct {
```

Comment:
And existing top-level objects just linger forever in etcd?

Reply:
Yes, unless the admins remove them before updating to a version where they can no longer be enabled.

Reply:
This is, sadly, the truth.

Reply:
More generally, probably remove any Pods that use classic DRA? Otherwise they might get treated as structured after the upgrade, leading to undefined behavior.

Reply:
Pods don't directly use classic DRA, but they might depend on claims which do. I've added:

ACTION REQUIRED: remove all workloads (pods, app deployments, etc. ) which depend on classic DRA and make sure that all PodSchedulingContext resources are gone before upgrading. PodSchedulingContext resources cannot be removed through the apiserver after an upgrade and workloads would not work properly.